runtime/interface.go

```diff
@@ -41,93 +43,124 @@ type Block struct {
 type ResolvedLocation = sema.ResolvedLocation
 type Identifier = ast.Identifier
 type Location = common.Location
-
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
-	// GetCode returns the code at a given location
-	GetCode(location Location) ([]byte, error)
-	// GetCachedProgram attempts to get a parsed program from a cache.
-	GetCachedProgram(Location) (*ast.Program, error)
-	// CacheProgram adds a parsed program to a cache.
-	CacheProgram(Location, *ast.Program) error
-	// GetValue gets a value for the given key in the storage, owned by the given account.
-	GetValue(owner, key []byte) (value []byte, err error)
-	// SetValue sets a value for the given key in the storage, owned by the given account.
-	SetValue(owner, key, value []byte) (err error)
-	// CreateAccount creates a new account.
-	CreateAccount(payer Address) (address Address, err error)
-	// AddAccountKey appends a key to an account.
-	AddAccountKey(address Address, publicKey []byte) error
-	// RemoveAccountKey removes a key from an account by index.
-	RemoveAccountKey(address Address, index int) (publicKey []byte, err error)
+type AddressLocation = common.AddressLocation
+
+// Accounts manages changes to the accounts, account keys and account storage
+//
+// errors returns by methods are non-fatal ones (e.g. accountNotExist),
+// fatal errors (e.g. Storage failure) are called by panic inside the methods
+// methods with power of edits have caller as an argument which can be used
+// to limit access to these method calls from smart contracts only, transaction only, or
+// specific smart contracts (e.g. service account).
+type Accounts interface {
```

Comment:
Would be nice to split the storage functions into a separate interface.

Reply:
Yeah, I tried that unfortunately there is a large dependency on the FVM side on the methods in this handler and since we would like to move to make keys to be resources, I think we can keep them here and long term remove them from this interface. what do you think ?

Reply:
Just split them into smaller ones, you're right it is more helpful on the cadence side to keep them separate, even if the same account provider on the fvm side provides all functionalities.